pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/ClassTypeResolver.java

```diff
@@ -335,13 +345,35 @@ public Object visit(ASTAnnotationTypeDeclaration node, Object data) {
 
     @Override
     public Object visit(ASTName node, Object data) {
```

Comment:
this method has now hit 100 lines. I guess it's time to refactor it into smaller pieces.